### ISSUE TITLE
diag.Diagnostics Warning is not displayed to the user

### ISSUE THREAD

**Issue by CyrusJavan** [hashicorp/terraform-plugin-sdk#703]
### SDK version

```
v2.4.3
```

### Relevant provider source code

```go
func resourceExamplePet() *schema.Resource {
	return &schema.Resource{
		Schema: map[string]*schema.Schema{
			"name": {
				Type:        schema.TypeString,
				Required:    true,
				Description: "Account name",
			},
		},

func resourceExamplePetCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
	var diags diag.Diagnostics
	diags = append(diags, diag.Diagnostic{
		Severity: diag.Warning,
		Summary:  "WARNING doing something weird here",
		AttributePath: cty.Path{
			cty.GetAttrStep{Name: "name"},
		},
	})

	c := meta.(*client.Pet).Pet
	petData := marshalExamplePet(d)

	resp, err := c.AddPet(&pet.AddPetParams{
		Body:    petData,
		Context: ctx,
	})

	if err != nil {
		return diag.Errorf("could not add pet: %v", err)
	}
	d.SetId(strconv.FormatInt(resp.Payload.ID, 10))

	return diags
}
```

### Terraform Configuration Files

```hcl
resource "example_pet" "pet_1" {
  name = "sharik"
}

```

### Debug Output

https://gist.github.com/CyrusJavan/6b87b69593003f031f7fb04d129baffb

### Expected Behavior

Terraform would display the warning to the user.
### Actual Behavior

No warning is displayed. It seems warnings are only displayed if one of the Diagnostic is an Error. This seems counter-intuitive to me.
### Steps to Reproduce

terraform apply

**Comment by paddycarver** [comment]
Can this still be reproduced against version 2.4.4 or later? We released a fix for something related at the time, and I think it probably fixed this, as well.

**Comment by paddycarver** [comment]
Sorry for the confusion, I'm thinking of another bug. I'm told this is a Terraform bug, not an SDK bug, and it has been resolved in 0.15.0.

**Comment by CyrusJavan** [comment]
Thanks for the info @paddycarver!

Could not reproduce the issue on v0.15.0.
